csharp/src/Google.Protobuf/Reflection/FieldDescriptor.cs

```diff
@@ -90,6 +90,12 @@ public sealed class FieldDescriptor : DescriptorBase, IComparable<FieldDescripto
         /// </summary>
         public override string Name { get { return proto.Name; } }
 
+
+        /// <summary>
+        /// The json_name option of the descriptor's target.
+        /// </summary>
+        public string JsonName { get { return proto.JsonName; } }
```

Comment:
Note that if I'm correct about protoc always populating the field, we may be able to just use the existing property, but still remove the conditions elsewhere.

This depends on the descriptors being generated with a recent version of protoc, of course. I'll check when all of this happened...